blog/content/post/2020-11-30-coreboot_for_power.md

```diff
@@ -0,0 +1,154 @@
+---
+title: coreboot port for OpenPOWER - why bother?
+abstract: 'You may have heard by now that we are working on coreboot port for
+          Talos II. OpenPOWER already has, nomen omen, open source firmware, so
+          one may ask why bother? We will try to answer that question.'
+cover: /covers/coreboot-logo.svg
+author: krystian.hebel
+layout: post
+published: true
+date: 2020-11-30
+archives: "2020"
+
+tags:
+  - firmware
+  - coreboot
+  - openpower
+categories:
+  - Firmware
+
+---
+
+You may have heard by now that we are working on coreboot port for [Talos II](https://raptorcs.com/TALOSII/).
+OpenPOWER already has, nomen omen, open source firmware, so one may ask why
+bother? In this blog post we will try to answer that question.
+
+## Short introduction to OpenPOWER boot process and its components
+
+Talos II is a server platform managed by a slightly customised (mainly to handle
+specific dual-CPU boot synchronisation quirks) [OpenBMC fork](https://git.raptorcs.com/git/talos-openbmc/).
+BMC signals the main (host) platform to start booting.
+
+Next comes the [Self-Boot Engine](https://wiki.raptorcs.com/wiki/Self-Boot_Engine).
+This term is actually used both for a chip (which is part of main CPU, it is a
+reduced PowerPC core) and its [firmware](https://git.raptorcs.com/git/talos-sbe/).
+SBE's tasks are initialisation of main CPU cores and loading and staring the
+next component called Hostboot.
+
+[Hostboot](https://git.raptorcs.com/git/talos-hostboot/) is the first component
+running on main cores. It is responsible mainly for the memory initialisation
+and training, it also initialises some of the platform buses.
+
+[Skiboot](https://git.raptorcs.com/git/talos-skiboot/) is then chainloaded. Its
+tasks include initialisation of the rest of the hardware, e.g. PCI Express host
+bus controllers. It also implements OPAL ([OpenPOWER Abstraction Layer](https://open-power.github.io/skiboot/doc/opal-spec.html#what-is-opal)),
+which is an interface to access firmware services from an OS. This is similar to
+the UEFI runtime services or legacy BIOS interrupt calls.
+
+Next link in this chain is [Skiroot](https://github.com/open-power/linux) (a
+small set of patches on top of the mainstream Linux kernel) accompanied by
+[Petitboot](https://git.raptorcs.com/git/talos-petitboot/) (userspace
+application). The latter is a bootloader, it uses kexec to load the final
+operating system from any device that can be mounted by Linux, including network
+boot.
+
+There are also other pieces of firmware running on different chips. Those are
+responsible for thermal and power management or CAPI (coherent accelerator
+processor interface) support. Most of them are loaded by the Hostboot.
```

Comment:
@krystian-hebel I counted quite a lot of those components, so I would highlight how, despite being open, OpenPOWER Firmware ecosystem is, whet inevitably have to lead to issues during maintenance.